sequencer/src/block.rs

```diff
@@ -917,19 +890,22 @@ mod test_headers {
 
     #[test]
     fn test_validate_proposal_error_cases() {
-        let (mut header, ..) = Header::genesis();
-        let mut block_merkle_tree = header.validated_state.block_merkle_tree.clone();
+        let (mut header, ..) = Header::genesis(&NodeState {});
+        let mut validated_state = ValidatedState::from_header(&header);
+        let mut block_merkle_tree = validated_state.block_merkle_tree.clone();
 
         // Populate the tree with an initial `push`.
         block_merkle_tree.push(header.commit()).unwrap();
```

Comment:
I think this is going to be a problem, because `ValidatedState::from_header` creates a partial view of the merklized state (just the root hashes, not any leaves) so we wouldn't be able to push to that. In actual operation we would have to fetch the missing state if we encountered this scenario. But for test purposes, we know what the state should be and so we might just have to create it separately from the header and assert that it's consistent.

Reply:
fixed in c8440bfff33f85094ae1635a4c0761a4be741527